validator/src/validation/length.rs

```diff
@@ -32,11 +42,126 @@ pub fn validate_length<T: HasLen>(
     true
 }
 
+pub trait ValidateLength {
+    fn validate_length(&self, min: Option<u64>, max: Option<u64>, equal: Option<u64>) -> bool;
+}
+
+impl ValidateLength for String {
+    fn validate_length(&self, min: Option<u64>, max: Option<u64>, equal: Option<u64>) -> bool {
+        validate_length_for_trait(self.chars().count() as u64, min, max, equal)
```

Comment:
What made you decide to go with `self.chars().count()` instead of `self.len()`?
And why use u64 everywhere instead of `usize`?

Reply:
I used `self.chars().count()` because the `HasLen` trait used them too and I didn't really think about it. The same is for `u64` instead of `usize`. The `validate_length` function used `Option<u64>`. I can refactor the trait to use `self.len()` and `usize` if it makes any difference.

Reply:
Strings should use `self.chars().count()`

Reply:
@Keats interesting, what is the advantage of that over `.len()`?

Reply:
```rs
fn main() {
    let txt = "日本語";
    println!("len {:?} vs chars {:?}", txt.len(), txt.chars().count());
}
```

will print

```
len 9 vs chars 3
```

In this case no one reasonable will think of len as number of bytes used (`.len()`) vs number of characters, which is what people think of when they think of a len for a string.

Reply:
Wow very interesting! Makes sense to use chars count then